client/gutenberg/extensions/markdown/editor.js

```diff
@@ -14,8 +13,10 @@ import './editor.scss';
 import edit from './edit';
 import save from './save';
 import { __ } from 'gutenberg/extensions/presets/jetpack/utils/i18n';
+import JetpackBlockType from 'gutenberg/extensions/presets/jetpack/utils/jetpack-block-type';
 
-registerBlockType( 'jetpack/markdown', {
+
+const MarkdownBlock = new JetpackBlockType( 'jetpack/markdown', {
```

Comment:
I must say I like the decentralized approach of registering blocks. The store I'm suggesting here could be the single point of centralization we need.